build/cli.ts

```diff
@@ -307,9 +307,12 @@ async function buildDocuments(
   }
 
   for (const [locale, meta] of Object.entries(metadata)) {
+    const sortedMeta = meta
+      .slice()
+      .sort((a, b) => a.mdn_url.localeCompare(b.mdn_url));
```

Comment:
The order of the `metadata.json` is non-deterministic even if you build a single locale, because the result of `fdir` is not deterministic, so sorting by locale is not enough, because pages within a locale would still be non-deterministic.

Note that I suggested to @fiji-flo to sort the result of `fdir`, but he pointed out that a non-deterministic build order is useful, because it prevents us from using side-effects of a deterministic build order.